components/Deck/ContentModulesPanel/DataSourcePanel/EditDataSource.js

```diff
@@ -7,10 +7,40 @@ import cancelEditDataSource from '../../../../actions/datasource/cancelEditDataS
 
 class EditDataSource extends React.Component {
     componentDidMount() {
+
+        $.fn.form.settings.rules.yearRule = (year) => {
+            let currDate = new Date();
+            let currYear = currDate.getYear() + 1900;;
```

Comment:
Date.getYear is deprecated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getYear , please use Date.getFullYear()